flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointRescaleITCase.java

```diff
@@ -61,7 +67,7 @@
         PIPELINE {
             @Override
             UnalignedSettings createSettings(int parallelism) {
-                int numShuffles = 8;
+                int numShuffles = 10;
```

Comment:
Can you extract this magic constant `10`? And maybe ideally calculate it automatically? Are we at least verifying it somewhere?

Reply:
Good idea to calculate it automatically. Follow-up PR?

Reply:
Removed it in the test refactoring commit. Please check if I should switch the order of the commits to avoid this change.